lib/Doctrine/DBAL/Query/QueryBuilder.php

```diff
@@ -501,21 +499,16 @@ public function distinct() : self
      *         ->leftJoin('u', 'phonenumbers', 'u.id = p.user_id');
      * </code>
      *
-     * @param mixed $select The selection expression.
+     * @param mixed $expression     The selection expressions.
+     * @param mixed ...$expressions Additional selection expressions.
      *
      * @return $this This QueryBuilder instance.
      */
-    public function addSelect($select = null) : self
+    public function addSelect($expression, ...$expressions) : self
     {
         $this->type = self::SELECT;
 
-        if (empty($select)) {
-            return $this;
-        }
-
-        $selects = is_array($select) ? $select : func_get_args();
-
-        return $this->add('select', $selects, true);
+        return $this->add('select', array_merge([$expression], $expressions), true);
```

Comment:
This is changing the behaviour since the first argument could be an array already.

Reply:
I should have probably changed the parameter type in the annotation from `mixed` to `string`. This is already reflected in the upgrade notes and there's another PR coming: https://github.com/morozov/dbal/compare/issues/3825...morozov:issues/3825-NEXT.

Reply:
For now, I've changed the annotated parameter type.